### PR TITLE
Don't escape HTML chars inside Django tags inside attribute values

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: hamlpy/parser/utils.py [hunk 1]
```diff
@@ -1,13 +1,32 @@
...
-def html_escape(s):
+DJANGO_TAG_OPEN = '{%'
+DJANGO_TAG_CLOSE = '%}'
+HTML_CHARS = {"&": "&amp;", "<": "&lt;", ">": "&gt;", '"': "&quot;", "'": "&#39;"}
+
+
+def html_escape(text):
     """
-    Escapes HTML entities, matching substitutions used the Ruby Haml library
+    Escapes HTML entities, matching substitutions used the Ruby Haml library. Entities that occur inside Django tags
```
Comment (nicpottier) [inline hamlpy/parser/utils.py]: "used by"

Comment (nicpottier) [inline hamlpy/parser/utils.py]: What about something like `{{ foo|escape:"&amp;" }}`?

**Comment by rowanseymour** [comment]
@nicpottier updated to also ignore entities inside {{ .. }} and fixed the typo

File: hamlpy/parser/utils.py [hunk 2]
```diff
@@ -1,13 +1,38 @@
...
+    or expressions are not escaped.
     """
-    s = s.replace("&", "&amp;")
-    s = s.replace("<", "&lt;")
-    s = s.replace(">", "&gt;")
-    s = s.replace('"', "&quot;")
-    s = s.replace("'", "&#39;")
-    return s
+    new_text = []
+    state = 0  # 0 = normal, 1 = in tag, 2 = in expression
```
Comment (nicpottier) [inline hamlpy/parser/utils.py]: should probably be enum ya?

Comment (nicpottier) [inline hamlpy/parser/utils.py]: if only there was a pep for that

**Pull request merged**